pkg/services/cosmosdb/cosmos-utils.go

```diff
@@ -178,3 +183,138 @@ func deleteDatabase(
+func (c *cosmosAccountManager) waitForReadRegionsReady(
+	ctx context.Context,
+	instance service.Instance,
+) (service.InstanceDetails, error) {
+	dt := instance.Details.(*cosmosdbInstanceDetails)
+	resourceGroupName := instance.ProvisioningParameters.GetString("resourceGroup")
+	accountName := dt.DatabaseAccountName
+	databaseAccountClient := c.databaseAccountsClient
+
+	err := pollingUntilReadRegionsReady(ctx, resourceGroupName, accountName, databaseAccountClient)
+	if err != nil {
+		return nil, err
+	}
+	return dt, nil
+}
+
+// For sqlAllInOneManager, the real type of `instance.Details` is `*sqlAllInOneInstanceDetails`,
+// so type assertion must be changed. Expect type assertion, this function is totally the same as previous one.
+// Do you have any good idea make the code cleaner?
+func (s *sqlAllInOneManager) waitForReadRegionsReady(
+	ctx context.Context,
+	instance service.Instance,
+) (service.InstanceDetails, error) {
+	dt := instance.Details.(*sqlAllInOneInstanceDetails)
+	resourceGroupName := instance.ProvisioningParameters.GetString("resourceGroup")
+	accountName := dt.DatabaseAccountName
+	databaseAccountClient := s.databaseAccountsClient
+
+	err := pollingUntilReadRegionsReady(ctx, resourceGroupName, accountName, databaseAccountClient)
+	if err != nil {
+		return nil, err
+	}
+	return dt, nil
+}
+
+// For now, this method will return on either context is cancelled or every region's state is "succeeded" in seven consecutive check.
+// The reason why we need seven consecutive check is that the read region is created one by one, there is a small gap between
+// the finishment of previous creation and the start of the next creation. By this check, we can detect gaps shorter than 1 mintue,
+// and report success within 70 seconds after completion.
+func pollingUntilReadRegionsReady(
+	ctx context.Context,
+	resourceGroupName string,
+	accountName string,
+	databaseAccountClient cosmosSDK.DatabaseAccountsClient,
+) error {
+	childCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	ticker := time.NewTicker(time.Second * 10)
+	previousSucceededTimes := 0
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			result, err := databaseAccountClient.Get(childCtx, resourceGroupName, accountName)
+			if err != nil {
+				return err
+			}
+			resultJSONBytes, err := json.Marshal(result)
+			if err != nil {
+				return err
+			}
+			resultJSONString := string(resultJSONBytes)
+
+			//Check whether every read location's state is "Succeeded"
+			allSucceed := true
+			readLocations := gjson.Get(resultJSONString, "properties.readLocations")
+			readLocations.ForEach(func(key, value gjson.Result) bool {
+				state := value.Get("provisioningState").String()
+				if state != "Succeeded" {
+					allSucceed = false
+					previousSucceededTimes = 0
+					return false
+				}
+				return true
+			})
+			if allSucceed && previousSucceededTimes >= 7 {
```

Comment:
looks like a magic number.
give it one comment why you choose 7, and make it a readable const variable name?

Reply:
[Here](https://github.com/norshtein/open-service-broker-azure/blob/region-in-template/pkg/services/cosmosdb/cosmos-utils.go#L223-L226) is the explanation.

Reply:
I will add detailed explanation and make it a readable const variable name later, Thanks!